src/main/java/com/facebook/openwifirrm/modules/ModelerUtils.java

```diff
@@ -202,4 +208,70 @@ public static double calculateTPCMetrics(
 			return Double.POSITIVE_INFINITY;
 		}
 	}
+
+	/**
+	 *
+	 * @param dataModel
+	 * @param obsoletionPeriodMs the maximum amount of time (in milliseconds) it is
+	 *                           worth aggregating over starting from the most
+	 *                           recent scan entry and working backwards in time
+	 * @param agg                an aggregator to output a calculated "aggregated
+	 *                           signal strength"
+	 * @return
+	 */
+	public Map<String, WifiScanEntryWrapper> getAggregatedWifiScans(Modeler.DataModel dataModel, long obsoletionPeriodMs,
```

Comment:
I think overall this function is great :). Thanks for improving it based on the suggestions. Just leave a small comment here for you to consider. 

1. I think latestWifiScans is sorted chronologically already (in a reverse order). 
2. Even if latestWifiScans was not sorted, I feel like we might not need to do sorting here. The current method includes the sorting (O(NlogN)) and the iteration (O(N) with an early stop so still the total complexity is O(NlogN)). If we don't do the sorting at all, the complexity is only O(N). Since we only save 10-20 scan results, the performance might not have any difference. Just FYI. 

Also, I would expect we will have some unit tests to cover this function. :) Thanks.

Reply:
latestWifiScans is sorted in that items are added as they are received. I wasn't sure that meant chronologically. E.g., what if (due to network issue, concurrency, anything) an earlier scan entry arrived later at the rrm code? Is that possible?

Reply:
Yes, I'm still working on writing unit tests...